tests/acceptance/acceptancesuite.js

```diff
@@ -160,6 +164,132 @@ test(`Facets load on the page, and can affect the search`, async t => {
+  // It should add 2 filter tags, one for the facet (from the backend)
+  // and one for the static filter we just clicked
+  await staticFilter.toggleOption('Frodo');
+  await t.expect(filterTags.count).eql(4);
+  await expectResultsCountToEql(2);
+  await t.expect(Selector('.yxt-StandardCard-title').nth(0).innerText).eql('Liz Frailey');
+  await t.expect(Selector('.yxt-StandardCard-title').nth(1).innerText).eql('Sophia Kleyman');
+
+  // Hit the back button, see the 'Frodo' filter disappear
+  await browserBackButton();
+  await t.expect(filterTags.count).eql(2);
+  await expectResultsCountToEql(1);
+  await t.expect(Selector('.yxt-StandardCard-title').innerText).eql('Liz Frailey');
+
+  // Hit the back button, see the 'Marty' filter disappear and return to the 'all' search.
+  await browserBackButton();
+  await t.expect(filterTags.count).eql(0);
+  await expectResultsCountToEql(initialResultsCount);
+
+  // Hit the forward button, should see the 'Marty' filter applied again
+  await browserForwardButton();
+  await t.expect(filterTags.count).eql(2);
+  await expectResultsCountToEql(1);
+  await t.expect(Selector('.yxt-StandardCard-title').innerText).eql('Liz Frailey');
+
+  // Hit the forward button, should see the 'Frodo' filter applied again
+  await browserForwardButton();
+  await t.expect(filterTags.count).eql(4);
+  await expectResultsCountToEql(2);
+  await t.expect(Selector('.yxt-StandardCard-title').nth(0).innerText).eql('Liz Frailey');
+  await t.expect(Selector('.yxt-StandardCard-title').nth(1).innerText).eql('Sophia Kleyman');
+});
+
+test(`filtersearch`, async t => {
+  const expectResultsCountToEql = async expectedCount => {
+    const currentResultsCount = await verticalResultsComponent.getResultsCountTotal();
+    await t.expect(currentResultsCount).eql(expectedCount);
+  };
+  const verticalResultsComponent = FacetsPage.getVerticalResultsComponent();
+  const filterSearch = FacetsPage.getFilterSearch();
+  const filterTags = Selector('.yxt-ResultsHeader-removableFilterTag');
+  await t.expect(Selector('.yxt-StandardCard-title').count).eql(0);
+
+  // Choose the 'Virginia, United States' filter option
+  await filterSearch.enterQuery('virginia');
+  await filterSearch.selectFilter('Virginia, United States');
+  await t.expect(filterTags.count).eql(1);
+  let filterTagText = await filterTags.nth(0).find('.yxt-ResultsHeader-removableFilterValue').innerText;
+  await t.expect(filterTagText).eql('Virginia, United States');
+  await expectResultsCountToEql(4);
+
+  // Choose the 'New York City, New York, United States' filter option
+  await filterSearch.enterQuery('new york');
+  await filterSearch.selectFilter('New York City, New York, United States');
+  await t.expect(filterTags.count).eql(1);
+  filterTagText = await filterTags.nth(0).find('.yxt-ResultsHeader-removableFilterValue').innerText;
+  await t.expect(filterTagText).eql('New York City, New York, United States');
+  await expectResultsCountToEql(1033);
```

Comment:
Is it possible to make these tests less dependent on the exact number of results? I know that we control the testing KG, but if we can test this without depending on the exact number I think that would be ideal. Similarly to my other comment, it is not very clear to me how some of these expect statements relate to the functionality that is being tested. Would explanatory boolean variables possibly work here and have the expects operate on those?

Reply:
originally I used a testcafe requestLogger to track that we were sending the correct filter in the request. BUT for some reason requestLogger wouldn't log the request from a page refresh, and only in browserstack (not locally or with the headless acceptance tests). I'll try to play around with it again in another branch because it definitely would be better to not care about the specific response

Reply:
Updated the tests to use RequestLogger! IE11 caches all ajax requests, which causes them to not be tracked by testcafe's RequestLogger (this was really annoying to uncover...), so I added a new IE11NoCacheHook which adds Cache-Control: no-store to the requests you tell the hook to care about